fileio/private/bids_sidecar.m

```diff
@@ -119,12 +121,16 @@
 % start with an empty return value
 sidecar = [];
 
+% if there is both a tsv and a json file for the desired sidecar, we want to return the tsv
```

Comment:
this is the most significant improvement for issue #1601 as it ensures that the tsv is read instead of the json